src/components/DeleteTemplate/DeleteTemplateButton.test.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  render,
+  screen,
+  fireEvent,
+  renderHook,
+  waitFor,
+} from '@testing-library/react';
+import { DeleteTemplateButton } from './DeleteTemplateButton';
+import { renderWithTheme } from '@/lib/test-utils';
+
+// Mocking translation and fetch utilities
+jest.mock('../../lib/i18n/client', () => ({
+  useTranslation: () => ({
+    t: (str: string) => str,
+    i18n: {
+      changeLanguage: () => new Promise(() => {}),
+    },
+  }),
+}));
+
+// Providing mock implementations for useRouter
+jest.mock('next/navigation', () => ({
+  useRouter: jest.fn().mockImplementation(() => ({
+    push: jest.fn(),
+    refresh: jest.fn(),
+  })),
+}));
+
+// Mocking the remove function
+jest.mock('../../lib/response/fetchUtil', () => ({
+  remove: jest.fn().mockResolvedValue({ status: 200 }),
+}));
+
+describe('DeleteTemplateButton', () => {
+  test('renders the button with the correct text', () => {
+    renderWithTheme(<DeleteTemplateButton templateId="1" lang="en" />);
+    const buttonElement = screen.getByTestId('DeleteTemplateButton');
+    expect(buttonElement).toBeInTheDocument();
+    expect(buttonElement).toHaveTextContent(
+      'DeleteTemplateButton.button.delete'
+    );
+  });
+});
+
+test('opens the popup when the button is clicked', () => {
```

Comment:
I think these should be under the describe('DeleteTemplateButton') function above?

Reply:
Fixed